src/main.rs

```diff
@@ -858,11 +900,15 @@ fn main() -> Result<(), Box<dyn Error>> {
         opts.address
     );
 
-    let runtime = tokio::runtime::Builder::new_multi_thread()
+    let rt = tokio::runtime::Builder::new_multi_thread()
         .worker_threads(10)
         .enable_all()
         .build()?;
-    runtime.block_on(async {
+
+    rt.block_on(async {
+        // Start metrics service
+        task::spawn(axum::Server::bind(&metrics_addr).serve(metrics_router.into_make_service()));
+
         let server = Server::bind(&opts.address).serve(service);
```

Comment:
@Daniel-Bloom-dfinity I would have liked to use `try_join!(...)` here to await on the metrics server as well, but I ran into some type issues. Maybe you would have an idea on how to do it and can show me?

Reply:
You could copy the Prometheus scrape endpoint from the control-plane: https://gitlab.com/dfinity-lab/public/ic/-/blob/master/rs/boundary_node/control_plane/src/main.rs#L189

Reply:
@mstein-dfinity This does work, I was just pointing out that it'd be useful to run all the futures under a `try_join!` so failures can bubble up.

Reply:
Any reason this doesn't work?
```
        // Start metrics service
        let metrics = axum::Server::bind(&metrics_addr).serve(metrics_router.into_make_service());

        let server = Server::bind(&opts.address).serve(service);
        
        try_join!(
            server,
            metrics,
        )?;
```